### DISCUSSION TITLE
If I change the name of the script hosted in Greasy Fork, will it affect the auto update in Violentmonkey?

### DISCUSSION THREAD

**Discussion by cyfung1031** [violentmonkey/violentmonkey#1890] [Q&A]
@tophf 

If my script is installed from Greasy Fork originally named as "My Script Hello"
then someday I changed the script from version 0.1.0 to 0.2.0 and make the metablock `@name` become "My Script World"

How it would affect the auto update for those people who have already installed My Script Hello 0.1.0?

Just become My Script World 0.2.0 and in the future My Script World 0.3.0 0.4.0 ..etc ?

**Comment by tophf** [comment]
It won't be a problem for them because the installed script has an internal `id` slot which stays the same on update.

But you must make sure that your `@name` + `@namespace` is unique, which you can do like this:
`@namespace https://greasyfork.org/users/123` where 123 is your greasyfork profile.
